src/audio/interface.rs

```diff
@@ -136,7 +136,7 @@ impl Interface {
             let tx0: usize = dma_index + skip.0;
             let tx1: usize = tx0 + 1;
 
-            let (y0, y1) = block[block_index];
+            let (y1, y0) = block[block_index];
```

Comment:
Good catch. Would you mind fixing it at the root - on line 123? That way we could keep 0 for left and 1 for right.

With that change the title of the commit/PR will need changing.

Reply:
@phoracek I played around with this a bit more and the audio inputs were also reversed, so I fixed that too.

But I don't understand why you originally swapped the ordering of the pairs for input and output?

For testing I used the following code to confirm that the left input is being routed to the left output on a Patch SM:

```rust
audio_interface
    .handle_interrupt_dma1_str1(|audio_buffer| {
        for frame in audio_buffer {
            *frame = (frame.0, 0.0);
        }
    })
    .unwrap();
```

Reply:
It was a bug carried over from https://github.com/antoinevg/daisy_bsp/blob/main/src/audio.rs#L252.

The line 123 now looks right. What I don't get is why the line 139 is the other way around? With your changes the left input goes to the left output. However, if I use this library only to output audio, this patch introduces a new bug where values written to even samples now end up in the right output instead of left. So while this solves one bug, it introduces another.

Does that make sense? I'm puzzled... but I only get to review these PRs at night, so I'm not at my sharpest at the moment. Feel free to call be out on getting this wrong.

Reply:
Maybe left and right are swapped in the hardware? Anyhow, if you could test that both of the cases mentioned above work as expected, then I'd be happy to merge it.

BTW what board are you testing this with?

Reply:
Scratch what I said before. I reviewed some of my applications using this library and I think I was just working around this bug by swapping left and right.

If you can confirm that this patched code now works both for passthrough and that sound sent to left samples ends up in the left audio output, then I will be happy to merge this PR. Sorry for the noise.

Reply:
@phoracek Thanks.

Yes, I can confirm both passthrough and audio sent directly to left & right channels is working correctly on a Daisy Patch SM 👍